packages/net/_test.pony

```diff
@@ -669,3 +673,124 @@ class _TestTCPProxyNotify is TCPConnectionNotify
 
   fun ref connect_failed(conn: TCPConnection ref) =>
     _h.fail_action("sender connect failed")
+
+class _TestTCPConnectionFailed is UnitTest
+  fun name(): String => "net/TCPConnectionFailed"
+
+  fun ref apply(h: TestHelper) =>
+    h.expect_action("connection failed")
+
+    let host = "127.0.0.1"
+    let port = "12345"
+
+    try
+      let connection = TCPConnection(
+        h.env.root as AmbientAuth,
+        object iso is TCPConnectionNotify
+          let _h: TestHelper = h
+
+          fun ref connected(conn: TCPConnection ref) =>
+            _h.log("log: connected to a non-existent host and port")
+            _h.fail("fail: connected to a non-existent host and port")
+            _h.complete(false)
+            conn.close()
+
+          fun ref connect_failed(conn: TCPConnection ref) =>
+            _h.log("log: connection failed")
+            _h.complete_action("connection failed")
+            _h.complete(true)
+        end,
+        host,
+        port)
+      h.long_test(20_000_000_000)
+      h.dispose_when_done(connection)
+    else
+      h.fail("error creating TCPConnection")
+      h.complete(false)
+    end
+
+actor _Connector
+  be connect(h: TestHelper, host: String, port: String) =>
+    try
+      let connection = TCPConnection(
+        h.env.root as AmbientAuth,
+        object iso is TCPConnectionNotify
+          let _hh: TestHelper = h
```

Comment:
why _hh instead of _h?

I did a double take when I saw `_hh`.

Reply:
Fixed